docs/book/src/component-config-tutorial/generate_docs.go

```diff
@@ -0,0 +1,196 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfig
+
+import (
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/spf13/afero"
+	pluginutil "sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+	"sigs.k8s.io/kubebuilder/v3/test/e2e/utils"
+)
+
+type Sample struct {
+	ctx *utils.TestContext
+}
+
+func NewSample(binaryPath, samplePath string) Sample {
+	log.Infof("Generating the sample context of component-config...")
+
+	ctx := newSampleContext(binaryPath, samplePath, "GO111MODULE=on")
+
+	return Sample{&ctx}
+}
+
+func newSampleContext(binaryPath string, samplePath string, env ...string) utils.TestContext {
+	cmdContext := &utils.CmdContext{
+		Env: env,
+		Dir: samplePath,
+	}
+
+	testContext := utils.TestContext{
+		CmdContext: cmdContext,
+		BinaryName: binaryPath,
+	}
+
+	return testContext
+}
+
+// Prepare the Context for the sample project
+func (sp *Sample) Prepare() {
+	log.Infof("destroying directory for sample project")
+	sp.ctx.Destroy()
+
+	log.Infof("refreshing tools and creating directory...")
+	err := sp.ctx.Prepare()
+
+	CheckError("creating directory for sample project", err)
+}
+
+func (sp *Sample) GenerateSampleProject() {
+	log.Infof("Initializing the project")
+	err := sp.ctx.Init(
+		"--domain", "tutorial.kubebuilder.io",
+		"--repo", "tutorial.kubebuilder.io/project",
+		"--license", "apache2",
+		"--owner", "The Kubernetes authors",
+		"component-config",
+	)
+	CheckError("Initializing the project", err)
+
+	log.Infof("Adding a new config type")
+	err = sp.ctx.CreateAPI(
+		"--group", "config",
+		"--version", "v2",
+		"--kind", "ProjectConfig",
+		"--resource", "--controller=false",
+		"--make=false",
+	)
+	CheckError("Creating the API", err)
+}
+
+func (sp *Sample) UpdateTutorial() {
+	// 1. generate controller_manager_config.yaml
+	var fs = afero.NewOsFs()
+	err := afero.WriteFile(fs, filepath.Join(sp.ctx.Dir, "config/manager/controller_manager_config.yaml"),
+		[]byte(`apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
+kind: ControllerManagerConfig
+metadata:
+  labels:
+    app.kubernetes.io/name: controllermanagerconfig
+    app.kubernetes.io/instance: controller-manager-configuration
+    app.kubernetes.io/component: manager
+    app.kubernetes.io/created-by: project
+    app.kubernetes.io/part-of: project
+    app.kubernetes.io/managed-by: kustomize
+health:
+  healthProbeBindAddress: :8081
+metrics:
+  bindAddress: 127.0.0.1:8080
+webhook:
+  port: 9443
+leaderElection:
+  leaderElect: true
+  resourceName: 80807133.tutorial.kubebuilder.io
+clusterName: example-test`), 0600,
+	)
+	CheckError("fixing controller_manager_config", err)
+
+	// 2. add componentconfig
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "PROJECT"),
+		"More info: https://book.kubebuilder.io/reference/project-config.html",
+		"\ncomponentConfig: true",
```

Comment:
We should not need to add this one. 
ComponentConfig should be true when we scaffold with `kubebuilder init --domain tutorial.kubebuilder.io --component-config`. If we are not making it true then we have a bug, and we can sort it out in a follow-up.